syntax = "proto3";
package merrellyze;

service EllipseSearcher {
    rpc EllipseSearch (EllipseSearchRequest) returns
    (EllipseSearchResponse) {}
}


message Image {
    int32 id = 1;
    string path = 3;
}

message CaptureJobRecord {
    int32 id = 1;
    repeated Image image = 2;
}

message EllipseSearchRequest {
    repeated CaptureJobRecord cjr = 1;
}


message Location {
    int32 x = 1;
    int32 y = 2;
}

message EllipseTag {
    int32 image_id = 1;
    Location start = 2;
    Location end = 3;
}

message EllipseSearchResponse {
    repeated EllipseTag tag = 1;
}